test/oauth2.test.js

```diff
@@ -90,7 +90,53 @@ describe('OAuth2Strategy', function() {
         }).to.throw(TypeError, 'OAuth2Strategy requires a authorizationURL option');
       });
     }); // with only a verify callback
-    
+
+    describe('with an optionsFunction and no sessionKey', function() {
+      it('should throw', function() {
+        expect(function() {
+          new OAuth2Strategy({
+            optionsFunction: () => { 
```

Comment:
```suggestion
            optionsFunction: function() { 
```

remove es6 syntax for <= node 3